src/index.ts

```diff
@@ -1,18 +1,61 @@
-import { Route, NavigationGuardNext } from 'vue-router'
+import { store } from '@/store'
+import {
+  NavigationGuard,
+  NavigationGuardNext,
+  RouteLocationNormalized,
+} from 'vue-router'
 
-export default (datas?: any) => async (to: Route, from: Route, next: NavigationGuardNext) => {
+// TODO: refactor 
+declare interface RouterMiddlewareContext {
```

Comment:
I think this approach is not true. Because this line restricts the developer from being sent custom data to the middleware.

Reply:
of course, need like `declare module 'name'`
for example vue router meta 
```
declare module 'vue-router' {
  interface RouteMeta {
    // is optional
    isAdmin?: boolean
    // must be declared by every route
    requiresAuth: boolean
  }
}
```